spec.bs

```diff
@@ -2741,6 +2777,59 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   [=automatic beacon data/once=] set to true.
 </div>
 
+<div algorithm=get-initiator-ancestor-root-origin>
+  To <dfn>get the initiator ancestor root origin</dfn> of a given {{Document}} |sourceDocument|:
+
+  1. Let |navigable| be |sourceDocument|'s [=node navigable=].
+
+  1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+     navigable=]:
+  
+     1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+  1. If |navigable| is null, return null.
+
+  1. Otherwise:
+
+     1. [=Assert=]: |navigable| is a [=fenced navigable container/fenced navigable=].
+
+     1. Set |navigable| to |navigable|'s [=navigable/unfenced parent=].
+
+     1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+        navigable=]:
+
+        1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+     1. If |navigable| is null, return null.
+
+     1. Otherwise, return |navigable|'s [=navigable/active document=]'s [=Document/origin=].
+
+  *The rest of this section is non-normative*.
```

Comment:
I'd get rid of this line, and just wrap the entire thing below in an example div, and put the HTML-ish-stuff below in a pre-html tag. See https://wicg.github.io/fenced-frame/#fenced-frame-src-example for an example. I think that would be more clear.

Reply:
Wrapped the section in an example div.